proto/src/determined/api/v1/master.proto

```diff
@@ -24,6 +24,22 @@ message SSOProvider {
   string type = 3;
 }
 
+// Active notice from the server admin.
+message MaintenanceMessage {
+  option (grpc.gateway.protoc_gen_swagger.options.openapiv2_schema) = {
+    json_schema: { required: [ "id", "message", "start_time", "end_time" ] }
+  };
+
+  // Unique ID.
+  int32 id = 1;
+  // Text content of message.
+  string message = 2;
+  // Time to begin showing message.
+  google.protobuf.Timestamp start_time = 3;
+  // Time to stop showing message.
+  google.protobuf.Timestamp end_time = 4;
```

Comment:
```suggestion
  optional google.protobuf.Timestamp end_time = 4;
```

Since `end_time` is optional, perhaps we should mark this field with `optional` for clarity.